src/App.tsx

```diff
@@ -15,35 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Default = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+interface FilterParams {
+  sortBy: SortBy;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortBy, isReversed }: FilterParams,
+): string[] {
+  const currentVisibleGoods = [...goods];
+
+  if (sortBy) {
+    currentVisibleGoods.sort((good1: string, good2: string) => {
+      if (sortBy === SortBy.Alphabetically) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortBy === SortBy.Length) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed) {
+    currentVisibleGoods.reverse();
+  }
+
+  return currentVisibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortByState, setSortByState] = useState(SortBy.Default);
+  const [reversedState, setReversedState] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
Boolean variables should start from `is` or `has`, and state in name is redundant